components/brave_wallet/browser/zcash/zcash_wallet_service.cc

```diff
@@ -626,14 +602,78 @@ void ZCashWalletService::CreateShieldAllTransaction(
   auto internal_addr = keyring_service_->GetOrchardRawBytes(
       account_id, CreateOrchardInternalKeyId(account_id));
 
-  auto shield_funds_task = base::WrapUnique<ZCashCreateShieldTransactionTask>(
-      new ZCashCreateShieldTransactionTask(
-          *this, chain_id, account_id, *internal_addr, std::nullopt,
-          kZCashFullAmount, std::move(callback)));
-  shield_funds_task->ScheduleWorkOnTask();
+  auto shield_funds_task =
+      std::make_unique<ZCashCreateTransparentToOrchardTransactionTask>(
+          base::PassKey<ZCashWalletService>(), *this,
+          CreateActionContext(account_id, chain_id), *internal_addr,
+          std::nullopt, kZCashFullAmount, std::move(callback));
+  shield_funds_task->Start();
   create_shield_transaction_tasks_.push_back(std::move(shield_funds_task));
 }
 
+void ZCashWalletService::CreateOrchardToOrchardTransaction(
+    const std::string& chain_id,
+    mojom::AccountIdPtr account_id,
+    const std::string& address_to,
+    uint64_t amount,
+    std::optional<OrchardMemo> memo,
+    CreateTransactionCallback callback) {
+  auto receiver_addr =
+      GetOrchardRawBytes(address_to, chain_id == mojom::kZCashTestnet);
+  if (!receiver_addr) {
+    std::move(callback).Run(
+        base::unexpected(l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR)));
+    return;
+  }
+
+  auto transfer_shield_funds_task =
+      std::make_unique<ZCashCreateOrchardToOrchardTransactionTask>(
+          base::PassKey<ZCashWalletService>(), *this,
+          CreateActionContext(account_id, chain_id), *receiver_addr,
+          std::move(memo), amount, std::move(callback));
+  transfer_shield_funds_task->Start();
+  create_shielded_transaction_tasks_.push_back(
+      std::move(transfer_shield_funds_task));
+}
+
+void ZCashWalletService::CreateTransparentToOrchardTransaction(
```

Comment:
There's some more CHECK usage here (and below this call) that we likely want to review and make sure is how we want it.

Reply:
Switched couple CHECKS to DCHECK. Other seems fine to me since they are critical.